RecoParticleFlow/PFProducer/src/PFAlgo.cc

```diff
@@ -1225,23 +1382,31 @@ void PFAlgo::processBlock( const reco::PFBlockRef& blockref,
 
 	// Check if this track is a muon
 	bool thatIsAMuon = PFMuonAlgo::isMuon(elements[jTrack]);
+        if (debug_) {
+            cout << " found track " << jTrack << (thatIsAMuon ? " (muon) " : " (non-muon)") << ", with distance = " << sortedECAL.begin()->first << std::endl;
+        }
 
 
 	// Check if this track is not a fake
 	bool rejectFake = false;
 	reco::TrackRef trackRef = elements[jTrack].trackRef();
 	if ( !thatIsAMuon && trackRef->ptError() > ptError_) { 
+          // GP: FIXME this selection should be adapted depending on hasDeadHcal
+          //     but we don't know if jTrack is linked to a dead Hcal or not
 	  double deficit = trackMomentum + trackRef->p() - clusterRef->energy();
 	  double resol = nSigmaTRACK_*neutralHadronEnergyResolution(trackMomentum+trackRef->p(),
 								    clusterRef->positionREP().Eta());
 	  resol *= (trackMomentum+trackRef->p());
-	  if ( deficit > nSigmaTRACK_*resol ) { 
+	  if ( deficit > nSigmaTRACK_*resol  && !goodTrackDeadHcal) { 
```

Comment:
isn't there a way to know?
Isn't ```hasDeadHcal``` valid/useful here?

perhaps I'm missing a full context.
This part for all-good hcal seems like an update/change to track selections

Reply:
`hasDeadHcal` and `goodTrackDeadHcal` are for the track being processed (`iTrack`). 
In order to figure out if `jTrack` is also attached to a dead hcal I would have to duplicate here the at least part of the logic of https://github.com/cms-sw/cmssw/blob/43fbb1c8f28f1608e78f73e3c9ba7ed190ae4b50/RecoParticleFlow/PFProducer/src/PFAlgo.cc#L990-L1065 ( `jTrack` can be `> iTrack`, and in that case the track hasn't gone through this logic)

To keep things simpler I avoided this, and just assumed that if `iTrack` is passing through a dead Hcal area, and `jTrack` is linked to the same Ecal element of `iTrack`, then it's very likely that also `jTrack` may have a bad Hcal, and in that case I don't want to reject it here.

In this loop anyway if the track as`jTrack` is not rejected as fake it's also not promoted to PFCandidate or set as inactive, and thus it will be processed eventually in further iterations of the main `iTrack` loop, and it can then be promoted or rejected.